homework/index.js

```diff
@@ -30,15 +35,97 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  function renderError(err) {
+    const root = document.getElementById('root');
+    createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  }
+
+  // deletes the previous repository table if selected repository changes
+  function removeChildElements(parent) {
+    while (parent.firstChild) {
+      parent.removeChild(parent.firstChild);
+    }
+  }
+
+  // to show the information of selected repository in a table
+  function showRepoInfo(index, repoDiv, repoArray, containerDiv, contributorsDiv) {
+    removeChildElements(repoDiv);
+    const selectedRepo = repoArray[index];
+    const headerArray = ['Repository', 'Description', 'Forks', 'Updated'];
+    const repoInfo = [
+      selectedRepo.name,
+      selectedRepo.description,
+      selectedRepo.forks,
+      selectedRepo.updated_at,
+    ];
+    const table = createAndAppend('table', repoDiv);
+    for (let i = 0; i < headerArray.length; i++) {
```

Comment:
You could combine the headers with values array in one, it makes it easier to maintain
Like,

    repoInfo = [
        { header: 'Repository', value: selectedRepo.name },
        { header: 'Description', value: selectedRepo.description },
        /* and so on */ 
    ];

Reply:
I changed the code as you mentioned and it is very handy advice, thank you.